tests/src/webhooks/handleInvoiceCompleted.test.ts

```diff
@@ -0,0 +1,365 @@
+      const getActiveSubscriptionsSpy = jest.spyOn(paymentService, 'getActiveSubscriptions');
+
+      await handleInvoiceCompleted(
+        fakeInvoiceCompletedSession,
+        usersService,
+        paymentService,
+        getLogger(),
+        cacheService,
+        config,
+        objectStorageService,
+      );
+
+      expect(getCustomerSpy).toHaveBeenCalledTimes(1);
+      expect(getInvoiceItemsSpy).toHaveBeenCalledTimes(1);
+      expect(getActiveSubscriptionsSpy).toHaveBeenCalledTimes(0);
+    });
+
+    it('When the price metadata has no maxSpaceBytes, then an error indicating so is thrown', async () => {
+      const mockedInvoice = getInvoice();
+      jest.spyOn(paymentService, 'getInvoiceLineItems').mockResolvedValue(mockedInvoice.lines.data as any);
+      const getActiveSubSpy = jest.spyOn(paymentService, 'getActiveSubscriptions');
+      if (mockedInvoice.lines.data[0].price && mockedInvoice.lines.data[0].price.metadata) {
+        mockedInvoice.lines.data[0].price.metadata = {};
+      }
+
+      const log = getLogger();
+
+      await handleInvoiceCompleted(
+        mockedInvoice,
+        usersService,
+        paymentService,
+        log,
+        cacheService,
+        config,
+        objectStorageService,
+      );
+
+      expect(getActiveSubSpy).not.toHaveBeenCalled();
+      expect(mockedInvoice.lines.data[0].price?.metadata.maxSpaceBytes).toBeUndefined();
+    });
+  });
+
+  describe('The subscription is for an object storage sub', () => {
+    it('When the product is not an obj storage type, then skips to the next process', async () => {
+      const mockedCustomer = getCustomer();
+      const mockedInvoice = getInvoice();
+      const mockedProduct = getProduct({});
+      const log = getLogger();
+
+      jest.spyOn(paymentService, 'getProduct').mockResolvedValue(mockedProduct as any);
+      const reactivateObjAccountSpy = jest.spyOn(objectStorageService, 'reactivateAccount').mockImplementation();
+
+      await handleObjectStorageInvoiceCompleted(
+        mockedCustomer,
+        mockedInvoice,
+        objectStorageService,
+        paymentService,
+        log,
+      );
+
+      expect(reactivateObjAccountSpy).not.toHaveBeenCalled();
+    });
+
+    it('When the invoice is completed, then the object storage account is activated', async () => {
+      const mockedCustomer = getCustomer();
+      const mockedInvoice = getInvoice();
+      const mockedProduct = getProduct({ userType: UserType.ObjectStorage });
+      const log = getLogger();
+
+      jest.spyOn(paymentService, 'getProduct').mockResolvedValue(mockedProduct as any);
+      const reactivateObjAccountSpy = jest.spyOn(objectStorageService, 'reactivateAccount').mockImplementation();
+
+      await handleObjectStorageInvoiceCompleted(
+        mockedCustomer,
+        mockedInvoice,
+        objectStorageService,
+        paymentService,
+        log,
+      );
+
+      expect(reactivateObjAccountSpy).toHaveBeenCalled();
```

Comment:
You should check that the params received on this function are the expected ones